app/com/linkedin/drelephant/spark/heuristics/ConfigurationParametersHeuristic.scala

```diff
@@ -0,0 +1,611 @@
+         driverEvaluator.severityDriverCores)
+
+       val driverScore = driverEvaluator.severityJvmUsedMemory.getValue +
+       driverEvaluator.severityDriverCores.getValue
+
+       (driverCores, driverMemory, driverSeverity, driverScore)
+     }
+   }
+
+   /**
+     * Calculate the severity for the stage level heuristics.
+     *
+     * @param analyses the analysis for all the stages
+     * @return the stage heristics severity.
+     */
+   private def calculateStageSeverity(analyses: Seq[StageAnalysis]): Severity = {
+     val stageSeverities = analyses.map(_.getStageAnalysisResults.maxBy(_.severity.getValue).severity)
+     if (stageSeverities.size > 0) {
+       stageSeverities.maxBy(_.getValue)
+     } else {
+       Severity.NONE
+     }
+   }
+
+   /**
+     * Calculate the recommended amount of memory, based on how much was used, and if there
+     * are changes to the number of cores (assume memory used is proportional to number of cores).
+     *
+     * @param allocatedMemory allocated memory
+     * @param jvmUsedMemory max JVM used memory
+     * @param numCores current number of cores
+     * @param recommendedCores recommended number of cores
+     * @return the recommended memory in bytes.
+     */
+   private def calculateRecommendedMemory(
+       allocatedMemory: Long,
+       jvmUsedMemory: Long,
+       numCores: Int,
+       recommendedCores: Int): Long = {
+     val calculatedMem = (jvmUsedMemory * Math.ceil(recommendedCores / numCores.toDouble) *
+       (1 + DEFAULT_MEMORY_BUFFER_PERCENTAGE) + JvmUsedMemoryHeuristic.reservedMemory).toLong
+     Math.min(Math.max(DEFAULT_MIN_MEMORY, calculatedMem), allocatedMemory)
+   }
+
+   /** If the severity is sigificant (not NONE or LOW). */
+   private def hasSignificantSeverity(severity: Severity): Boolean = {
+     severity != Severity.NONE && severity != Severity.LOW
+   }
+
+   /**
+     * Calculate the total amount of unified memory allocated across all tasks for
+     * the stage.
+     *
+     * @param executorMemory executor memory in bytes
+     * @param memoryFraction spark.memory.fraction
+     * @param executorCores number of executor cores
+     * @param numTasks number of tasks/partitions
+     * @return
+     */
+   private def calculateTotalUnifiedMemory(
+       executorMemory: Long,
+       memoryFraction: Double,
+       executorCores: Int,
+       numTasks: Int): Long = {
+     // amount of unified memory available for each task
+     val unifiedMemPerTask = (executorMemory - SPARK_RESERVED_MEMORY) * memoryFraction /
+       executorCores
+     (unifiedMemPerTask * numTasks).toLong
+   }
+
+   private val REGEX_MATCHER = raw"(\d+)((?:[T|G|M|K])?B?)?"r
+
+   /**
+     * Given a memory value in bytes, convert it to a string with the unit that round to a >0 integer part.
+     *
+     * @param size The memory value in long bytes
+     * @return The formatted string, null if
+     */
+   private def bytesToString(size: Long): String = {
```

Comment:
This method is available in org.apache.spark.network.util.JavaUtils.

Reply:
I've replaced JstringToBytes with avaUtils.byteStringAsBytes. I wasn't able to find a function for the other direction.